pkg/fanal/analyzer/language/python/pip/pip.go

```diff
@@ -41,3 +96,152 @@ func (a pipLibraryAnalyzer) Type() analyzer.Type {
+	if err != nil {
+		return "", err
+	}
+	pythonExecDir := filepath.Dir(pythonExecPath)
+
+	// Search for a directory starting with "python" in the lib directory
+	libDir := filepath.Join(pythonExecDir, "..", "lib")
+	spDir, err := findSitePackagesDir(libDir)
+	if err != nil {
+		return "", xerrors.Errorf("Unable to detect `site-packages` dir for %q: %w", pythonExecPath, err)
+	}
+	if spDir != "" {
+		return spDir, nil
+	}
+
+	// Try another common pattern if the Python library directory is not found
+	spDir = filepath.Join(pythonExecDir, "..", "..", "lib", "site-packages")
+	if _, err = os.Stat(spDir); os.IsNotExist(err) {
+		return "", xerrors.Errorf("site-packages directory not found")
+	}
+
+	return spDir, nil
+}
+
+// pythonExecutablePath returns path to Python executable
+func pythonExecutablePath() (string, error) {
+	for _, execName := range pythonExecNames {
+		// Get the absolute path of the python command
+		pythonPath, err := exec.LookPath(execName)
+		if err != nil {
+			continue
+		}
+		return pythonPath, nil
+	}
+	return "", xerrors.Errorf("Unable to find path to Python executable")
+}
+
+// findSitePackagesDir finds `site-packages` dir in `lib` dir
+func findSitePackagesDir(libDir string) (string, error) {
+	entries, err := os.ReadDir(libDir)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return "", xerrors.Errorf("failed to read lib directory: %w", err)
+		}
+		return "", nil
+	}
+
+	// Use latest python dir
+	var spDir string
+	for _, pythonDir := range sortPythonDirs(entries) {
+		// Found a directory starting with "python", assume it's the Python library directory
+		dir := filepath.Join(libDir, pythonDir, "site-packages")
+		if _, err = os.Stat(dir); !os.IsNotExist(err) {
+			spDir = filepath.Join(libDir, pythonDir, "site-packages")
+		}
+
+	}
+	return spDir, nil
+}
+
+// sortPythonDirs finds `python` dirs and sorts them according to major and minor versions.
+// e.g. python2.7 => python3.9 => python3.11
+func sortPythonDirs(entries []os.DirEntry) []string {
+	var pythonDirs []string
+	for _, entry := range entries {
+		if entry.IsDir() && strings.HasPrefix(entry.Name(), "python") {
+			pythonDirs = append(pythonDirs, entry.Name())
+		}
+	}
+
+	sort.Slice(pythonDirs, func(i, j int) bool {
+		majorVer1, minorVer1, found1 := strings.Cut(pythonDirs[i], ".")
+		majorVer2, minorVer2, found2 := strings.Cut(pythonDirs[j], ".")
+
+		// if one of dir doesn't contain minor version => sort as strings
+		if !found1 || !found2 {
+			return pythonDirs[i] < pythonDirs[j]
+		}
+
+		// Sort major versions as strings
+		// e.g. `python2.7` and `python3.10`
+		if majorVer1 != majorVer2 {
+			return majorVer1 < majorVer2
+		}
+
+		// Convert minor versions
+		ver1, err1 := strconv.Atoi(minorVer1)
+		ver2, err2 := strconv.Atoi(minorVer2)
+
+		// If we can't convert minor versions => sort as strings
+		if err1 != nil || err2 != nil {
+			return pythonDirs[i] < pythonDirs[j]
+		}
+
+		return ver1 < ver2
+	})
```

Comment:
I believe go-version helps.
https://go.dev/play/p/Ytgevsntqr7

Reply:
hm... i didn't think about this 👍 
Updated in https://github.com/aquasecurity/trivy/pull/6782/commits/226b3dac9e64dec6acf557166dd6c6d06d1009e4